daffodil-cli/src/it/scala/org/apache/daffodil/CLI/Util.scala

```diff
@@ -173,9 +173,13 @@ object Util {
     val expectedCode = expectedExitCode.id
 
     val keyWord = "EXITCODE:"
-    val exitCodeCmd = "echo " + keyWord + (if (Util.isWindows) "%errorlevel%" else "$?")
+//    val exitCodeCmd = "echo " + keyWord + (if (Util.isWindows) "%errorlevel%" else "$?")
+
+    val exitCodeCmd = "echo " + keyWord + (if (Util.isWindows) "^|%errorlevel%" else "\\!$?")
```

Comment:
Please remove the commented code. Also, I think it's worth adding a comment explaining why we have this `^|` and `\\!` stuff. The idea being that the echo command needs to look different from what the echo outputs or the expect command might match the actual command. By echo things that are escaped (an escaped pipe in windows and an escaped exclamation in linux), then the echo command and the echo output include different text, so expect can tell the difference.